app/logical/danger_zone.rb

```diff
@@ -5,6 +5,14 @@ def self.uploads_disabled?(user)
     user.level < min_upload_level
   end
 
+  def self.post_visible?(post, user)
+    if hide_pending_posts_for <= 0
+      return true
+    end
+
+    (!user.nil? && user.is_approver?) || !post.is_pending? || post.created_at.before?(hide_pending_posts_for.hours.ago)
```

Comment:
all staff should bypass this, not just approvers
we specifically have an `is_staff?` function that can be used

Also in normal usage CurrentUser *should* never be nil, the only places it can possibly be nil are in the console or tests

https://github.com/e621ng/e621ng/blob/2ba1173d36326ef902790b12baea3241227ad732/app/controllers/application_controller.rb#L9
https://github.com/e621ng/e621ng/blob/2ba1173d36326ef902790b12baea3241227ad732/app/controllers/application_controller.rb#L166-L168
https://github.com/e621ng/e621ng/blob/2ba1173d36326ef902790b12baea3241227ad732/app/logical/session_loader.rb#L17

Reply:
> Also in normal usage CurrentUser should never be nil, the only places it can possibly be nil are in the console or tests

Yeah it was my mistake, I switched away from it in a leter commit

Reply:
> we specifically have an `is_staff?` function that can be used

I'll add it, but since not all approvers are staff because the permission can be granted separately, it'll need both